litestar/app.py

```diff
@@ -826,14 +825,6 @@ def update_openapi_schema(self) -> None:
 
         operation_ids: list[str] = []
 
-        if not self._openapi_schema.components:
-            self._openapi_schema.components = Components()
-            schemas = self._openapi_schema.components.schemas = {}
-        elif not self._openapi_schema.components.schemas:
-            schemas = self._openapi_schema.components.schemas = {}
-        else:
-            schemas = {}
```

Comment:
The thing I didn't understand here was the case where the openapi config object defined a `components` object with a `schemas` dict that had values in it. In that case, we'd drop into the `else` clause and use a dict that isn't bound to the openapi object at all.

Reply:
I think that was quite simply a bug.

Reply:
yep, definitely looks that way!